R/data.table.R

```diff
@@ -2360,8 +2360,8 @@ split.data.table <- function(x, f, drop = FALSE, by, sorted = FALSE, keep.by = T
     join = TRUE
   }
   dtq[["j"]] = substitute(
-    list(.ll.tech.split=list(.expr)),
-    list(.expr = if (join) quote(if(.N == 0L) .SD[0L] else .SD) else as.name(".SD")) # simplify when `nomatch` accept NULL #857 ?
+    list(.ll.tech.split=list(.expr), .ll.tech.split.names=paste(lapply(.BY, as.character), collapse=".")),
```

Comment:
no risk of collision here? `by = c('x', 'y')` when the table has `x.y` as a column?

Reply:
the risk is pretty low, requires columns to be named `.ll.tech.split` or `.ll.tech.split.names`